backend/src/main/java/gov/cdc/usds/simplereport/service/TestOrderService.java

```diff
@@ -416,32 +421,32 @@ public AddTestResultResponse addMultiplexResult(
     return new AddTestResultResponse(savedOrder, deliveryStatus);
   }
 
-  private void editMultiplexResult(TestOrder order, List<MultiplexResultInput> newResults) {
-    // For now - we still need to save the covid results to the result column
-    // To be removed in #3664
-    Optional<MultiplexResultInput> covidResult =
-        newResults.stream().filter(r -> r.getDiseaseName().equals("COVID-19")).findFirst();
-    covidResult.ifPresent(diseaseResult -> order.setResultColumn(diseaseResult.getTestResult()));
-
+  private Set<Result> editMultiplexResult(TestOrder order, List<MultiplexResultInput> newResults) {
     // For new results, check if there's already a pending result for the same test.
     // If so, update it, if not, create a new one.
-    newResults.forEach(
-        newResult -> {
-          Optional<Result> pendingResult =
-              _resultRepo.getPendingResult(
-                  order, _diseaseService.getDiseaseByName(newResult.getDiseaseName()));
-          if (pendingResult.isPresent()) {
-            pendingResult.get().setResult(newResult.getTestResult());
-            _resultRepo.save(pendingResult.get());
-          } else {
-            Result result =
-                new Result(
-                    order,
-                    _diseaseService.getDiseaseByName(newResult.getDiseaseName()),
-                    newResult.getTestResult());
-            _resultRepo.save(result);
-          }
-        });
+    return newResults.stream()
+        .map(
+            newResult -> {
+              Optional<Result> pendingResult =
+                  _resultRepo.getPendingResult(
+                      order, _diseaseService.getDiseaseByName(newResult.getDiseaseName()));
+              if (pendingResult.isPresent()) {
+                pendingResult.get().setResult(newResult.getTestResult());
+                order.addResult(pendingResult.get());
```

Comment:
Does this just get us around the initialization issue with `order.getResults()`? `_resultRepo.save(pendingResult)` should be enough to persist the result in the database with a link to the testOrder, so I'm curious what this extra call does.